src/Manifolds.jl

```diff
@@ -42,6 +9,7 @@ import ManifoldsBase:
     array_value,
     base_manifold,
     check_manifold_point,
+    check_manifold_point__transparent,
```

Comment:
why do we have to explicitly import transparent functions?

Reply:
I don't know. These were imported in `MetricManifold.jl`, so I moved them here. If they're not necessary, I could remove them (one is explicitly extended).

Reply:
So `exp!__intransparent` is imported but not extended, so that can be left out, but I don't recall why we did that with `inner`, but it doesn't look nice. I have to check whether that's necessary.

Reply:
and for `check_manifold_point__` I don't see where that's used?

Reply:
Ah, I see what the issue is. `@decorator_transparent_fallback` creates a method with a name like `exp!__intransparent`, but since that particular function is also created by `@decorator_transparent_fallback` in ManifoldsBase, it needs to be imported. There must be a more hygienic  way to do this that doesn't require the user to know the internals of the macro in ManifoldsBase and whether the macro has been called there.

Reply:
That's beyond the scope of this issue, so I'll leave the imports in.

Reply:
Yes, leave it imported for now then. I didn't notice this when @mateuszbaran moved to changing the dispatch-per-symbol to different functions, but we should look for a – as you said – more hygenic way to do this and leave the internals “hidden”.

It might be that just for this one case its required and it might also be solved by moving MetricManifold to base (which I would actually favour, including moving basic metric types) – as well as maybe moving power and product... but thats all beyond this PR for sure ;)